e2e/framework/server_utils.go

```diff
@@ -0,0 +1,108 @@
+package framework
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+// serveAgent will be a simple http file server that will expose our
+// freshly compiled devpod binaries to be downloaded as agents.
+// useful for non-linux runners
+func ServeAgent() {
+	// Specify the directory containing the files you want to serve
+	dir := "bin"
+
+	wd, err := os.Getwd()
+	if err == nil {
+		dir = filepath.Join(wd, "bin")
+	}
+
+	// Create a file server handler for the specified directory
+	fileServer := http.FileServer(http.Dir(dir))
+
+	// Register the file server handler to serve files under the /files route
+	http.Handle("/files/", http.StripPrefix("/files", fileServer))
+
+	randomPort, err := findOpenPort(0)
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Is there a hard constraint on having a port between 10k and 40k?

Otherwise, we could replace this port-finding process with a call to `listener, err := net.Listen("tcp", fmt.Sprintf("%v:0", ip))`.
That should return a listener that `http.Serve` can consume, and we can grab the port by using `listener.Addr().(*net.TCPAddr).Port`

Reply:
No constraints, indeed your suggested solution simplifies things a lot, thanks for pointing it out! :smile:

Reply:
Done :+1: